ext/phalcon/mvc/url/utils.c

```diff
@@ -189,7 +189,11 @@ void phalcon_replace_paths(zval *return_value, zval *pattern, zval *paths, zval
 	unsigned int bracket_count = 0, parentheses_count = 0, intermediate = 0;
 	unsigned char ch;
 	smart_str route_str = {0};
+#if PHP_VERSION_ID < 70000
 	ulong position = 1;
+#else
+	zend_ulong position = 1;
+#endif
```

Comment:
The files in ext are generated by Zephir. It isn't useful to edit these files. 
Phalcon 3.4 isn't compatible with 7.4. Bigger changes are needed to support this and that isn't something that is currently on our roadmap.

Reply:
@ruudboon As I remember this file isn't generated by Zephir. So all ok with this https://github.com/phalcon/cphalcon/blob/c493def8747a711ee879c86eb1459f18df770409/config.json#L36

Reply:
Ah yeah, missed that that one is external. Still not sure if we should merge this since we're not supporting 7.4 in the 3.4 branch. We could enable the 7.4 tests to see what happens.

Reply:
@ruudboon agree, I also want to see what happens if enable php 7.4 tests. It seems ok on my mac:
```
➜  cphalcon git:(3.4.x) ✗ php -v
PHP 7.4.1 (cli) (built: Dec 18 2019 14:46:18) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.1, Copyright (c), by Zend Technologies


➜  cphalcon git:(3.4.x) ✗ php --ri=phalcon

phalcon


Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 3.4.5
Build Date => Jan  7 2020 15:26:48
Powered by Zephir => Version 0.12.15-$Id$

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off
phalcon.orm.update_snapshot_on_save => On => On
phalcon.orm.disable_assign_setters => Off => Off
```